### ISSUE TITLE
Provide ability to skip current request using postman.setNextRequest()

### ISSUE THREAD

**Issue by kurtr** [a85/POSTMan-Chrome-Extension#7325]
I've found many related issues that would solve this issue such as request tags etc but non with an identical implementation. Basically it all comes down to running tests for a single collection with multiple enviroment files for dev/uat/prod etc.

Currently we mostly use folders for versioning but it would be great to simple be able to use something like:

    if(env != 'dev'){
       pm.setNextRequest(pm.getNextRequest())
    }

or simply have `pm.setNextRequest()` skip the current request if no args are provided.

**Comment by dseligson** [comment]
Yes--This would be very helpful!
We have a scenario where some Consumers are not allowed access to all APIs.
Given one of these test runs, an error message (API_NOT_ALLOWED) would be Successful.
But then I would want to skip the rest of the API calls thereafter such that:
if (API_NOT_ALLOWED) { pm.skipCurrentRequest(); }

**Comment by arnaudlacour** [comment]
friendly bump, this suggestion would make programmatic workflows a lot cleaner.

**Comment by dseligson** [comment]
me 2!  Bumpity Bump

**Comment by sri-gutala-sage** [comment]
This feature will be very helpful !!

**Comment by dandiep** [comment]
would really like this feature but without custom coding. E.g. we have a failing test due to a bug. Would be great to just disable the request via a checkbox

**Comment by stefanluchian** [comment]
Yes, it would be very nice to be able to skip the current step i.e. the current request and its tests and let Postman decide which is the next step.

**Comment by bean5** [comment]
I agree that this is important. I imagine that the traditional "Use `setNextRequest()`" advice started before we had pre-request scripts. But that's just a guess. This single feature was enough to make me just use jmeter instead for quite a while. But I recently switched because other features became good enough to merit a switch.

But `setNextRequest()` still smells like GOTO from BASIC days. It is brittle, hard to follow, and easily missed.

**Comment by bean5** [comment]
`setNextRequest()` with no args...interesting idea. But would be hard to follow. To me it would look like code that would crash. I'd prefer `skipRequest()` as an option in pre-request scripts while `setNextRequest()` is an option in post-request scripts (ie tests or otherwise).

**Comment by ylishevvia** [comment]
If you have 30 requests in a collection you need to keep track of the ids in case you want to run all of them except for one or two by using the setNextRequest. It would be better if you have a natural order (position in the list) and skip the current request. setNextRequest potentially could accept an integer number where 1 is the next request, 2 is the request after the next one, -1 is the last one, -2 is the request before the last one and so on.

In addition using "tags" would be great too. To execute a set of request based on a tag, but I guess that is way more work than adding a way to move to the next request.

Third idea could be to have a context - useful, probably lazy loaded information. For example it would be great to have the current collection available with the meta information, so knowing the current request you could find which is the next by looking at the collection. Right now I don't see how I can do that, except probably passing the name of the collection and the folder from the command line and then load the collection from FS.

**Comment by mlafleur** [comment]
A simple `skipRequest()` would be fantastic. We have numerous tests that only run in certain environments. This results in a slew of individual tests being wrapped by `Pre_Test` and `Post_Test` (wrapped to avoid the fragility of `setNextRequest('name')`).

**Comment by w136712058** [comment]
So , is there any oppertunity to support skipRequst() in PreTest?

**Comment by TLYoz** [comment]
+1 There are many similar requests to this all pretty much the same all generating a lot of interest.

**Comment by kluceroInspirato** [comment]
This would be a great feature. Not having to explicitly define the next test to execute in order to prevent execution of the current request would add a lot of durability to tests.

**Comment by as-ajitsingh** [comment]
Bumping this, Seems like a very useful feature.

**Comment by hilr** [comment]
add this line `pm.request.url = '';` in pre-request scripts will help you skip sending the request.

**Comment by Yelenanis1** [comment]
@hilr this would throw an error and stop the collection execution after this request.

**Comment by vedkribhu** [comment]
We have introduced the api `pm.execution.skipRequest()` to allow skipping current request execution. You can read more about it here: https://blog.postman.com/optimize-your-api-testing-workflow-with-postmans-skip-request-feature/
